src/main/java/io/aiven/kafka/connect/http/sender/SslContextBuilder.java

```diff
@@ -0,0 +1,115 @@
+ */
+
+package io.aiven.kafka.connect.http.sender;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509ExtendedTrustManager;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.Socket;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+import io.aiven.kafka.connect.http.config.HttpSinkConfig;
+
+final class SslContextBuilder {
+
+    private static final TrustManager DUMMY_TRUST_MANAGER = new X509ExtendedTrustManager() {
+        @Override
+        public void checkClientTrusted(final X509Certificate[] chain, final String authType, final Socket socket)
+            throws CertificateException {
+        }
+
+        @Override
+        public void checkServerTrusted(final X509Certificate[] chain, final String authType, final Socket socket)
+            throws CertificateException {
+        }
+
+        @Override
+        public void checkClientTrusted(final X509Certificate[] chain, final String authType, final SSLEngine engine)
+            throws CertificateException {
+        }
+
+        @Override
+        public void checkServerTrusted(final X509Certificate[] chain, final String authType, final SSLEngine engine)
+            throws CertificateException {
+        }
+
+        @Override
+        public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+            return new java.security.cert.X509Certificate[0];
+        }
+
+        @Override
+        public void checkClientTrusted(final X509Certificate[] chain, final String authType)
+            throws CertificateException {
+        }
+
+        @Override
+        public void checkServerTrusted(final java.security.cert.X509Certificate[] chain, final String authType)
+            throws CertificateException {
+        }
+    };
+
+    static SSLContext createSslContext(final HttpSinkConfig config) 
+            throws NoSuchAlgorithmException, KeyManagementException {
+        final SSLContext sslContext = SSLContext.getInstance("TLS");
+        if (config.sslTrustAllCertificates()) {
+            sslContext.init(null, new TrustManager[] {DUMMY_TRUST_MANAGER}, new SecureRandom());
+        } else {
+            final TrustManagerFactory tmf = loadTruststore(config);
+            sslContext.init(null, tmf != null ? tmf.getTrustManagers() : null, new SecureRandom());
+        }
+        return sslContext;
+    }
+
+    private static TrustManagerFactory loadTruststore(final HttpSinkConfig config) {
+        if (config.sslTruststoreLocation() == null) {
+            return null;
+        }
+        try {
+            final KeyStore trustStore = KeyStore.getInstance("JKS");
```

Comment:
Would it be a good idea to also add a config for the [keystore type](https://docs.oracle.com/en/java/javase/21/docs/specs/security/standard-names.html#keystore-types)?

Reply:
Correct. This is only for server validation not for client validation. But makes me think if mTLS might be a future feature request. Anyway, that is out of scope for this PR.